src/main/java/duke/data/task/Task.java

```diff
@@ -0,0 +1,32 @@
+package duke.data.task;
+
+public abstract class Task {
+    protected String description;
+    protected boolean isDone;
+
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
+    }
+
+    public void markAsDone(){
+        isDone = true;
+    }
+    public abstract String getType();
+    public abstract String store();
+    public String getStatusIcon() {
+        return (isDone ? "X" : " "); // mark done task with X
```

Comment:
Avoid using magic strings like this - extract them out as constants and name them using proper naming conventions